azure-pipelines.yml

```diff
@@ -80,6 +80,20 @@ jobs:
     condition: eq(variables['Build.Reason'], 'PullRequest')
     displayName: 'Use clang-format to check C/C++ coding style'
 
+  - bash: |
+      fork_origin="$(git merge-base --fork-point origin/master)"
+      changed_files="$(git diff --name-only "$fork_origin")"
+      if [[ -z "$changed_files" ]]; then
+        xargs util/fix_include_guard.py --dry-run <<< "$changed_files" | tee fix-include-guard-output
+        if [[ -s fix-include-guard-output ]]; then
+          echo -n "##vso[task.logissue type=error]"
+          echo "Include guard check failed. Please run util/fix_include_guard.py on the above files."
+          exit 1
+        fi
+      fi
+    condition: eq(variables['Build.Reason'], 'PullRequest')
```

Comment:
You don't need to limit this check to changed files, it should be idempotent.

Reply:
I did this mostly so that changes that don't touch C at all don't have to wait on this check. As our number of headers increases the check will be slower.